File: PHP/MaximizeTheConfusionOfAnExamTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MaximizeTheConfusionOfAnExamTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MaximizeTheConfusionOfAnExam.php");

class MaximizeTheConfusionOfAnExamTest extends TestCase {

    public function testMaximizeTheConfusionOfAnExam() {
        $obj = new MaximizeTheConfusionOfAnExam();
        //Test 1
        $want = 4;
        $got = $obj->maxConsecutiveAnswers( "TTFF", 2 );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 3;
        $got = $obj->maxConsecutiveAnswers( "TFFT", 1 );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = 5;
        $got = $obj->maxConsecutiveAnswers( "TTFTTFTT", 1 );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MaximizeTheConfusionOfAnExam_test.go
/**
 * @version GO 1.20.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMaximizeTheConfusionOfAnExam
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMaximizeTheConfusionOfAnExam(t *testing.T) {
	tables := []struct {
		want   int
		answer string
		k      int
	}{
		{4, "TTFF", 2},
		{3, "TFFT", 1},
		{5, "TTFTTFTT", 1},
	}

	for _, table := range tables {
		got := maxConsecutiveAnswers(table.answer, table.k)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
